libethashseal/Ethash.cpp

```diff
@@ -152,8 +152,13 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 	{
 		if (_bi.number() >= chainParams().u256Param("EIP158ForkBlock"))
 		{
-			int chainID(chainParams().u256Param("chainID"));
-			_t.checkChainId(chainID);
+			if (_bi.number() >= chainParams().u256Param("metropolisForkBlock") && _t.hasZeroSignature())
+				_t.checkChainId(0);
```

Comment:
@winsvega if it throws, you have incorrect chainId in your `v` in zero signature